loaf-explainer.md

```diff
@@ -0,0 +1,345 @@
+
+            // when the function was invoked. Note that this is the startTime of the script, not
+            // the startTime of the frame (each entry in the performance timeline has a startTime)
+            startTime,
+
+            // If this script was parsed/compiled, this would be the time after compilation.
+            // Otherwise it would be equal to startTime
+            executionStart,
+
+            // the duration between startTime and when the subsequent microtask queue has finished
+            // processing
+            duration,
+
+            // Total time spent in forced layout/style inside this function
+            forcedStyleAndLayoutDuration,
+
+            // The time when the callback was queued, e.g. the event timeStamp or the time when
+            // the timeout was supposed to be invoked.
+            desiredExecutionStart,
+
+            // In the case of promise resolver this would be the invoker's source location
+            sourceLocation: "functionName@URL:line:col",
+
+            // Relationship between the (same-origin) window where this script was executed and
+            // this window.
+            windowAttribution: "self" | "descendant" | "ancestor" | "same-page" | "other"
+
+            // A reference to the same-origin window that originated the script, if it's still
+            // alive.
+            window,
+        }
+    ]
+}
+```
+
+### Processing model
+
+The new proposal:
+
+```js
+
+let frameTiming = null;
+
+while (true) {
+    if (frameTiming === null) {
+        frameTiming = new AnimationFrameTiming();
+        frameTiming.startTime = performance.now();
+    }
+
+    const task = eventQueue.pop();
+    if (task)
+        task.run();
+
+    if (!hasDocumentThatNeedsRender()) {
+        frameTiming.renderEnd = performance.now();
+        if (frameTiming.renderEnd - frameTiming.startTime > 50)
+            reportLongAnimationFrame();
+        frameTiming = null;
+        continue;
+    }
+
+    if (!hasRenderingOpportunity())
+        continue;
+
+    invokeAnimationFrameCallbacks();
+    frameTiming.styleAndLayoutStart = performance.now();
+    for (const document of documentsInThisEventLoop) {
+        while (document.needsStyleOrLayout()) {
+            document.calculateStyleAndLayout();
+            invokeResizeObserverCallbacks();
+        }
+    }
+    frameTiming.renderEnd = performance.now();
+    markPaintTiming();
+    if (frameTiming.renderEnd - frameTiming.StartTime > 50)
+        reportLongAnimationFrame();
+
+    frameTiming = null;
+    render();
+}
+```
+
+### Security & Privacy Considerations
+
+At the most part, LoAF only exposes information across same-origin windows. Information about
+scripts within a window is already observable, e.g. using resource timing or a service worker.
+
+However, LoAF might expose rendering information for a particular document tree that may be
+cross-origin (same-agent/site). The details about rendering the frame, such as
```

Comment:
Is this statement relying on cross-site isolation?

Reply:
Given that the rest of the paragraph talks about agents, maybe drop the "site" bit?